app/src/ui/changes/sidebar.tsx

```diff
@@ -76,8 +76,8 @@ export class ChangesSidebar extends React.Component<IChangesSidebarProps, {}> {
 
   private receiveProps(props: IChangesSidebarProps) {
     if (
-      props.repository.id !== this.props.repository.id ||
-      !this.autocompletionProviders ||
+      this.autocompletionProviders === null ||
+      props.repository !== this.props.repository ||
```

Comment:
I'd love to understand what the actual fix here is, as we're no longer checking `props.repository.id` here which might be a different equality check here.

Reply:
so the problem was that the `id` check doesn't capture situations where the repository `id` hasn't changed, but its properties have (like `Repository.GitHubRepository` going from `null` to something).

so, once a repository with a null `GitHubRepository` has been shown in the sidebar, you would have to trigger one of these conditions to get coauthoring to work in the UI.

Reply:
Cool, that's good to know.

What if we compared `.hash` instead, which also touches `.gitHubRepository`?

https://github.com/desktop/desktop/blob/9c9c44fb9128edce2821e01310b9ad06a635d52a/app/src/models/repository.ts#L47-L54

Reply:
i love that idea

Reply:
f2823c7